cmd/bucket-replication.go

```diff
@@ -1666,7 +1666,7 @@ func replicateObjectWithMultipart(ctx context.Context, c *minio.Core, bucket, ob
 		cHeader := http.Header{}
 		cHeader.Add(xhttp.MinIOSourceReplicationRequest, "true")
 		if !isSSEC {
-			crc := getCRCMeta(objInfo, partInfo.Number, nil) // No SSE-C keys here.
+			crc := getCRCMeta(objInfo, partIdx+1, nil) // No SSE-C keys here.
```

Comment:
we should always use partInfo.Number - is it because this is not populated?

Reply:
yes

Reply:
that is strange, it can be pre-populated and there is no reason to make this change here.

Reply:
hash.ReadCheckSums is already extracting the right part checksum given the part number. Don't see the need to prepopulate this or save it in xl.meta

Reply:
No, I meant whoever returns you PartInfo must already prepopulate this, it doesn't need to be persisted. My point is this is not clean, the code that exists now is clean. 

just prepopulate it when ToObjectInfo() happens.

Reply:
because this nuance is unnatural from upper layers to understand. It is safe to add PartNumber such that other callers (there may be others using this would not bother with it to deal with loop index numbers)

Reply:
AFAICT there can be gaps in part numbers, so this does not appear safe. Can you identify in which cases the number is missing and maybe fill it in then?

Reply:
eventually, they are all sorted @klauspost and written so its always part.1 is 1st element. If we write checksum encryption based on the final index, then there is no issue.

Reply:
there is already a bug with hash.ReadCheckSums because it assumes the parts are sequential. If a CompleteMultipart finalizes parts 4 and 7 - a subsequent GET for that part(4 or 7) with checksum enabled returns InvalidPartNumber error, whereas asking for part number 1 or 2 returns the checksum properly.

Reply:
@poornas yeah we need to fix all of that properly.